docs/topics/item-pipeline.rst

```diff
@@ -163,6 +166,52 @@ method and how to clean up the resources properly.
 .. _MongoDB: https://www.mongodb.org/
 .. _pymongo: https://api.mongodb.org/python/current/
 
+
+Take screenshot of item
+-----------------------
+
+This example demonstrates how to return Deferred_ from :meth:`process_item` method.
+It uses Splash_ to render screenshot of item url. Pipeline
+makes request to locally running instance of Splash_. After request is downloaded
+and Deferred callback fires, it saves item to a file and adds filename to an item.
+
+::
+
+    import scrapy
+
+
+    class ScreenshotPipeline(object):
+        """Pipeline that uses Splash to render screenshot of
+        every Scrapy item."""
+
+        SPLASH_URL = "http://localhost:8050/render.png?url={}"
+
+        def process_item(self, item, spider):
+            item_url = item["url"]
+            screenshot_url = self.SPLASH_URL.format(item_url)
+            request = scrapy.Request(screenshot_url)
+            dfd = spider.crawler.engine.download(request, spider)
+            dfd.addBoth(self.return_item, item)
+            return dfd
```

Comment:
I'm not sure, maybe it is possible, but the goal of example is to show how to return Deferred from process_item. Example with inline callbacks would not demonstrate this clearly, it would return generator decorated with `@inlineCallbacks`, we would also have to explain what are inline callbacks, link to some docs for them.

Other comments see #2287